samza-shell/src/main/bash/run-class.sh

```diff
@@ -45,19 +45,68 @@ DEFAULT_LOG4J2_FILE=$base_dir/lib/log4j2.xml
 BASE_LIB_DIR="$base_dir/lib"
 # JOB_LIB_DIR will be set for yarn container in ContainerUtil.java
 # for others we set it to home_dir/lib
-JOB_LIB_DIR="${JOB_LIB_DIR:-$base_dir/lib}"
+JOB_LIB_DIR="${JOB_LIB_DIR:-$home_dir/lib}"
 
 export JOB_LIB_DIR=$JOB_LIB_DIR
 
-echo JOB_LIB_DIR=$JOB_LIB_DIR
-echo BASE_LIB_DIR=$BASE_LIB_DIR
+# Usage: relative_path_from_to $source $target
+# Purpose: Given two absolute paths return relative path from $source to $destination
+function relative_path_from_to() {
+ source=${1%/} # This is $home_dir for this case
+ target=${2%/} # This is $BASE_LIB_DIR where JARS are
+
+ common_part=$source # start with common_part as source and iterate until common_part becomes $target
+ result=""
+
+ # This calculates the longest common prefix
+ while [[ "${target#$common_part}" == "${target}" ]]; do
+     # no match, means that candidate common part is not correct
+     # go up one level (reduce common part)
+     common_part="$(dirname $common_part)"
+     # and record that we went back, with correct / handling
+     if [[ -z $result ]]; then
+         result=".."
+     else
+         result="../$result"
+     fi
+ done
+
+ if [[ $common_part == "/" ]]; then
+     # special case for root (no common path)
+     result="$result/"
+ fi
+
+ # since we now have identified the common part,
+ # compute the non-common part
+ forward_part="${target#$common_part}"
+
+ # and now stick all parts together
+ if [[ -n $result ]] && [[ -n $forward_part ]]; then
+     result="$result$forward_part"
+ elif [[ -n $forward_part ]]; then
+     # extra slash removal
+     result="${forward_part:1}"
+ fi
+
+ echo "./$result"
+}
+
+# Get the relative path from current execution directory to base directory with jars
+RELATIVE_PATH_BASE_LIB=$(relative_path_from_to $home_dir $BASE_LIB_DIR)
+# If the relative path directory does not exist, resolve to absolute path
+if [ ! -d "$RELATIVE_PATH_BASE_LIB" ]; then
```

Comment:
it would be safer to check for existence of a jar file (samza-api\*.jar for example).

Reply:
will do